src/client/dfs/dfs_internal.h

```diff
@@ -337,20 +346,21 @@ oid_gen(dfs_t *dfs, daos_oclass_id_t oclass, bool file, daos_obj_id_t *oid)
 
 	D_MUTEX_LOCK(&dfs->lock);
 	/** If we ran out of local OIDs, alloc one from the container */
-	if (dfs->oid.hi >= MAX_OID_HI) {
+	if (dfs->oid.hi == 0) {
```

Comment:
I guess this isn't necessarily better...also need to start at a random place potentially